lib/reqd.ml

```diff
@@ -192,6 +192,30 @@ let send_fixed_response t s response data =
   | Fixed _ | Complete _ ->
     failwith "h2.Reqd.respond_with_*: response already complete"
 
+let send_trailers_on_close t new_trailers =
+  let go s =
+    match s.response_state with
+    | Streaming (rsp, rsp_body, _old_trailers) ->
```

Comment:
thoughts on failing if you try to schedule multiple times? I think this behavior should be somewhat specified, and additionally documented in the `.mli`

Reply:
I can't think of a use case where we'd need to change the trailers after setting them, and we should notify the user somehow that something is probably wrong.

I was hoping we could somehow avoid runtime errors, but I'm not sure how to do it with the current API. Maybe something to think about if we decide to change the API one day (we could then probably enforce it at the type level).

I'll error out for now.